src/__tests__/ceramic_integration.test.ts

```diff
@@ -0,0 +1,330 @@
+        ceramicPort: daemonPort2,
+        ganachePort,
+        port: casPort2,
+      })
+    await cas2.start()
+
+    // Finally make the Ceramic nodes that will be used in the tests.
+    ceramic1 = await makeCeramicCore(ipfs5, 'http://localhost:' + casPort1, ganacheURL)
+    ceramic2 = await makeCeramicCore(ipfs6, 'http://localhost:' + casPort2, ganacheURL)
+
+    // The two user-facing ceramic nodes need to have the same DID Provider so that they can modify
+    // each others streams.
+    const did = makeDID()
+    await did.authenticate()
+    ceramic1.did = did
+    ceramic2.did = did
+  });
+
+  afterAll(async () => {
+    cas1.stop();
+    cas2.stop();
+    await Promise.all([dbConnection1.close(), dbConnection2.close()])
+    await Promise.all([daemon1.close(), daemon2.close()])
+    await Promise.all([casCeramic1.close(), casCeramic2.close(), ceramic1.close(), ceramic2.close()])
+    await Promise.all([ipfsServer1.stop(), ipfsServer2.stop()])
+    await Promise.all([ipfs1.stop(), ipfs2.stop(), ipfs3.stop(), ipfs4.stop(), ipfs5.stop(), ipfs6.stop()])
+    await ganacheServer.close()
+  });
+
+  beforeEach(async () => {
+    console.log(`Starting test: ${expect.getState().currentTestName}`);
+  })
+
+  afterEach(async () => {
+    console.log(`Finished test: ${expect.getState().currentTestName}`);
+  })
+
+  test('Anchors are independent', async () => {
+    const doc1 = await TileDocument.create(ceramic1, {foo: 1}, null, { anchor: true })
+    const doc2 = await TileDocument.create(ceramic2, {foo: 1}, null, { anchor: true })
+
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+    expect(doc2.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+
+    // Test that anchoring on CAS1 doesn't anchor requests made against CAS2
+    await anchorUpdate(doc1, cas1)
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+    expect(doc2.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+
+    // Now test that anchoring on CAS2 doesn't anchor requests made against CAS1
+    await doc1.update({foo:2}, null, { anchor: true })
+    await anchorUpdate(doc2, cas2)
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+    expect(doc2.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+  }, 60 * 1000 * 3);
+
+  test('Multiple anchors for same stream', async () => {
+    const doc1 = await TileDocument.create(ceramic1, {foo: 1}, null, { anchor: true })
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+    await anchorUpdate(doc1, cas1)
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+
+    // Now that genesis commit has been anchored do an update and make sure anchoring works again
+    await doc1.update({foo:2}, null, { anchor: true })
+    await anchorUpdate(doc1, cas1)
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+    expect(doc1.content).toEqual({foo:2})
+  }, 60 * 1000 * 3);
+
+  test('Consensus for anchors', async () => {
```

Comment:
Can we rename this from `Consensus for anchors` to `Anchors latest update at time anchor is requested`?